spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jRepositoriesAutoConfiguration.java

```diff
@@ -16,45 +16,29 @@
 
 package org.springframework.boot.autoconfigure.data.neo4j;
 
-import org.neo4j.ogm.session.Neo4jSession;
+import org.neo4j.driver.Driver;
 
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
-import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
-import org.springframework.data.neo4j.repository.Neo4jRepository;
 import org.springframework.data.neo4j.repository.config.EnableNeo4jRepositories;
-import org.springframework.data.neo4j.repository.config.Neo4jRepositoryConfigurationExtension;
-import org.springframework.data.neo4j.repository.support.Neo4jRepositoryFactoryBean;
 
 /**
- * {@link EnableAutoConfiguration Auto-configuration} for Spring Data's Neo4j
- * Repositories.
- * <p>
- * Activates when there is no bean of type {@link Neo4jRepositoryFactoryBean} configured
- * in the context, the Spring Data Neo4j {@link Neo4jRepository} type is on the classpath,
- * the Neo4j client driver API is on the classpath, and there is no other configured
- * {@link Neo4jRepository}.
- * <p>
- * Once in effect, the auto-configuration is the equivalent of enabling Neo4j repositories
- * using the {@link EnableNeo4jRepositories @EnableNeo4jRepositories} annotation.
+ * Shared entry point for the configuration of Spring Data Neo4j repositories in their
+ * imperative and reactive forms.
  *
  * @author Dave Syer
  * @author Oliver Gierke
  * @author Josh Long
- * @since 1.4.0
+ * @author Michael J. Simons
  * @see EnableNeo4jRepositories
+ * @since 1.4.0
  */
 @Configuration(proxyBeanMethods = false)
-@ConditionalOnClass({ Neo4jSession.class, Neo4jRepository.class })
-@ConditionalOnMissingBean({ Neo4jRepositoryFactoryBean.class, Neo4jRepositoryConfigurationExtension.class })
-@ConditionalOnProperty(prefix = "spring.data.neo4j.repositories", name = "enabled", havingValue = "true",
-		matchIfMissing = true)
-@Import(Neo4jRepositoriesRegistrar.class)
+@ConditionalOnClass(Driver.class)
 @AutoConfigureAfter(Neo4jDataAutoConfiguration.class)
+@Import({ Neo4jImperativeRepositoriesConfiguration.class, Neo4jReactiveRepositoriesConfiguration.class })
 public class Neo4jRepositoriesAutoConfiguration {
```

Comment:
You could keep that type and create another auto-configuration for the reactive variant. That would be more consistent with other stores.